var/spack/repos/builtin/packages/xdmf3/package.py

```diff
@@ -30,8 +31,9 @@ class Xdmf3(CMakePackage):
     # See https://github.com/spack/spack/pull/22303 for reference
     depends_on(Boost.with_default_variants)
     depends_on("mpi", when="+mpi")
-    depends_on("hdf5+mpi", when="+mpi")
-    depends_on("hdf5~mpi", when="~mpi")
+    depends_on("hdf5@1.10:+mpi", when="+mpi")
+    depends_on("hdf5@1.10:~mpi", when="~mpi")
+    patch("fix_hdf5_hid_t.diff")
```

Comment:
Can you add a brief comment above the `patch` directive explaining:
1. What is the effect of the patch
2. Why it is needed
3. Where it was taken from

?

Reply:
It seems I totally forgot about this PR.
the patch is motivated by the discussion in https://gitlab.kitware.com/xdmf/xdmf/-/issues/28
In particular:
` in the HDF5 1.10-series, hid_t is defined int64_t`
The built was also complaining about the missing headers that I added.